From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Juan Sanchez Fuentes <juanesf91@gmail.com>
Date: Fri, 8 May 2026 01:07:46 -0400
Subject: Add regulator-always-on for wifi on sun55i-a527-cubie-a5e.dts

Adapted from Armbian build@0648ff3c4125d673c18b5f032dc7c28545c542b5,
patch/kernel/archive/sunxi-6.18/patches.backports/34-fix-wifi-regulator-cubie-a5e.patch.
Keep the kernel and firmware descriptions paired: Linux normally uses
the firmware DT to preserve SID-derived Ethernet MAC addresses.

Signed-off-by: Juan Sanchez Fuentes <juanesf91@gmail.com>
---
 dts/upstream/src/arm64/allwinner/sun55i-a527-cubie-a5e.dts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dts/upstream/src/arm64/allwinner/sun55i-a527-cubie-a5e.dts b/dts/upstream/src/arm64/allwinner/sun55i-a527-cubie-a5e.dts
index 111111111111..222222222222 100644
--- a/dts/upstream/src/arm64/allwinner/sun55i-a527-cubie-a5e.dts
+++ b/dts/upstream/src/arm64/allwinner/sun55i-a527-cubie-a5e.dts
@@ -263,6 +263,7 @@ reg_aldo4: aldo4 {
 			};
 
 			reg_bldo1: bldo1 {
+				regulator-always-on;
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <1800000>;
 				regulator-name = "vcc-pg-iowifi";
-- 
Armbian

